### ISSUE TITLE
delete from admin site list raises error

### ISSUE THREAD

**Issue by jacobg** [potatolondon/djangae#508]
If you delete one or models from a list on the admin site, this error occurs:

```
Traceback (most recent call last):
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/contrib/admin/options.py", line 583, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/utils/decorators.py", line 105, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/views/decorators/cache.py", line 52, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/contrib/admin/sites.py", line 206, in inner
    return view(request, *args, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/utils/decorators.py", line 29, in _wrapper
    return bound_func(*args, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/utils/decorators.py", line 105, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/utils/decorators.py", line 25, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/contrib/admin/options.py", line 1527, in changelist_view
    response = self.response_action(request, queryset=cl.get_queryset(request))
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/contrib/admin/options.py", line 1282, in response_action
    response = func(self, request, queryset)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/contrib/admin/actions.py", line 49, in delete_selected
    queryset.delete()
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/db/models/query.py", line 557, in delete
    collector.delete()
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/db/models/deletion.py", line 262, in delete
    qs._raw_delete(using=self.using)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/db/models/query.py", line 569, in _raw_delete
    sql.DeleteQuery(self.model).delete_qs(self, using)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/db/models/sql/subqueries.py", line 85, in delete_qs
    self.get_compiler(using).execute_sql(NO_RESULTS)
  File "/Users/j/Documents/workspace/f-gae-dev/src/django/db/models/sql/compiler.py", line 776, in execute_sql
    sql, params = self.as_sql()
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/compiler.py", line 49, in as_sql
    return (DeleteCommand(self.connection, self.query), tuple())
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/commands.py", line 902, in __init__
    self.select = SelectCommand(connection, query, keys_only=True)
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/commands.py", line 393, in __init__
    self.query = transform_query(connection, query)
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/query.py", line 1153, in transform_query
    return _FACTORY[version](connection, kind, query)
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/query.py", line 1044, in _transform_query_17
    model=query.model
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/query.py", line 906, in _walk_django_where
    walk_node(query.where, **kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/query.py", line 896, in walk_node
    leaf_callback(child, **new_kwargs)
  File "/Users/j/Documents/workspace/f-gae-dev/src/djangae/db/backends/appengine/query.py", line 934, in _django_17_query_walk_leaf
    raise NotSupportedError("Attempted to run a subquery on the datastore")
NotSupportedError: Attempted to run a subquery on the datastore
```

**Comment by nealedj** [comment]
This issue seems unlikely to be specific to Django admin and is probably more of an issue with the deletion collector's behaviour with the datastore query limitations.

I think we need some more information to action a fix here. The query that is being run will depend on the relationships defined in the app's models.

**Comment by adamalton** [comment]
Possibly half workaround: set `on_delete=models.DO_NOTHING` in your field definition, which should then stop it trying to collect all the things before deleting.  But obviously that also changes the behaviour.  Just thought it was worth mentioning in case it's a viable workaround for anyone.

Will discuss possible solutions in #800.

**Comment by Kazade** [comment]
Fixed in #800
